src/ripple/conditions/impl/Der.cpp

```diff
@@ -0,0 +1,782 @@
+TraitsCache::length(void const* addr, size_t l)
+{
+    if (lengthCache_.empty())
+        lengthCache_.reserve(32);
+    lengthCache_[addr] = l;
+}
+
+boost::optional<boost::container::small_vector<size_t, 8>>
+TraitsCache::sortOrder(void const* addr) const
+{
+    auto const i = sortOrderCache_.find(addr);
+    if (i != sortOrderCache_.end())
+        return i->second;
+    return boost::none;
+}
+
+void
+TraitsCache::sortOrder(
+    void const* addr,
+    boost::container::small_vector<size_t, 8> const& so)
+{
+    if (sortOrderCache_.empty())
+        sortOrderCache_.reserve(32);
+    sortOrderCache_[addr] = so;
+}
+
+//------------------------------------------------------------------------------
+
+Tag::Tag(SequenceTag)
+    : classId(ClassId::universal), tagNum(16), primitive(false)
+{
+}
+
+Tag::Tag(SetTag)
+    : classId(ClassId::universal), tagNum(17), primitive(false)
+{
+}
+
+bool
+Tag::isSet() const
+{
+    return classId == ClassId::universal && tagNum == 17;
+}
+
+//------------------------------------------------------------------------------
+
+std::uint64_t
+tagNumLength(std::uint64_t v)
+{
+    if (v <= 30)
+        return 1;
+
+    std::size_t n = 1 + 8 * sizeof(v) / 7;
+
+    // skip leading zeros
+    while (n--)
+    {
+        auto b = static_cast<std::uint8_t>((v >> (n * 7)) & 0xFF);
+        if (b)
+            break;
+    }
+
+    assert(n);
+    return n + 2;
+}
+
+void
+encodeTagNum(MutableSlice& dst, std::uint64_t v, std::error_code& ec)
+{
+    assert(v > 30);
+    std::size_t n = 1 + 8 * sizeof(v) / 7;
+
+    // skip leading zeros
+    while (n--)
+    {
+        auto b = static_cast<std::uint8_t>((v >> (n * 7)) & 0xFF);
+        if (b)
+            break;
+    }
+
+    assert(n);
+    ++n;
+    if (dst.size() != n)
+    {
+        ec = make_error_code(Error::logicError);
+        return;
+    }
+    while (n--)
+    {
+        auto b = static_cast<std::uint8_t>((v >> (n * 7)) & 0xFF);
+        // all but the last byte has the high order bit set
+        if (n)
+            b |= 1 << 7;
+        else
+            b &= ~(1 << 7);
+        dst.push_back(static_cast<char>(b));
+    }
+}
+
+void
+encodeContentLength(MutableSlice& dst, std::uint64_t v, std::error_code& ec)
+{
+    if (v <= 127)
+    {
+        if (dst.size() != 1)
+        {
+            ec = make_error_code(Error::logicError);
+            return;
+        }
+        dst.push_back(static_cast<char>(v));
+        return;
+    }
+
+    std::size_t n = sizeof(v);
+
+    // skip leading zeros
+    while (n--)
+    {
+        auto b = static_cast<std::uint8_t>((v >> (n * 8)) & 0xFF);
+        if (b)
+            break;
+    }
+
+    ++n;
+    assert(n);
+
+    if (dst.size() != n + 1)
+    {
+        ec = make_error_code(Error::logicError);
+        return;
+    }
+
+    dst.push_back(static_cast<char>(n) | (1 << 7));
+
+    while (n--)
+        dst.push_back(static_cast<char>((v >> (n * 8)) & 0xFF));
+}
+
+size_t
+contentLengthLength(std::uint64_t v)
```

Comment:
@mellery451 Thanks. I just pushed a patch that should fix this (changed decl return type from `size_t` to `uint64_t`.